logos/logos-workernode/logos_worker_node/calibration.py

```
@@ -3152,105 +3152,117 @@ def calibrate_with_tp_escalation(
+        # Binary search: try progressively smaller tp values.
+        # tp must be a power of 2 in vLLM, so we halve each step.
+        low_tp = original_tp
+        high_tp = tp
+        while low_tp < high_tp:
+            if cancel_event is not None and cancel_event.is_set():
+                break
+            mid_tp = high_tp // 2
+            if mid_tp < low_tp:
+                break
+            logger.info(
+                "  %s trying tp=%d (search range %d–%d)",
```

Ruff (0.16.3)

[warning] String contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF001)